sql3/planner/inbuiltfunctionsstring.go

```diff
@@ -644,3 +646,82 @@ func (n *callPlanExpression) EvaluateReplicate(currentRow []interface{}) (interf
 	return strings.Repeat(stringArg, int(intArg)), nil
 
 }
+
+func (p *ExecutionPlanner) analyseFunctionStr(call *parser.Call, scope parser.Statement) (parser.Expr, error) {
+	if len(call.Args) < 1 || len(call.Args) > 3 {
+		return nil, sql3.NewErrCallParameterCountMismatch(call.Rparen.Line, call.Rparen.Column, call.Name.Name, 1, len(call.Args))
+	}
+	targetType := parser.NewDataTypeDecimal(4)
+	if !typesAreAssignmentCompatible(targetType, call.Args[0].DataType()) {
+		return nil, sql3.NewErrTypeAssignmentIncompatible(call.Args[0].Pos().Line, call.Args[0].Pos().Column, call.Args[0].DataType().TypeDescription(), targetType.TypeDescription())
+	}
+
+	// the second and third parameters are optional
+	for i := 1; i < len(call.Args); i++ {
+		if !typeIsInteger(call.Args[i].DataType()) {
+			return nil, sql3.NewErrIntExpressionExpected(call.Args[i].Pos().Line, call.Args[i].Pos().Column)
+		}
+	}
+
+	call.ResultDataType = parser.NewDataTypeString()
+
+	return call, nil
+}
+
+func (n *callPlanExpression) EvaluateStr(currentRow []interface{}) (interface{}, error) {
+	argOneEval, err := n.args[0].Evaluate(currentRow)
+	if err != nil {
+		return "", err
+	}
+	if argOneEval == nil {
+		return nil, nil
+	}
+
+	coercedValue, err := coerceValue(n.args[0].Type(), parser.NewDataTypeDecimal(4), argOneEval, parser.Pos{})
+	if err != nil {
+		return "", err
+	}
+
+	decimalArgOne, ok := coercedValue.(pql.Decimal)
+	if !ok {
+		return nil, sql3.NewErrInternalf("unexpected type conversion '%T'", coercedValue)
```

Comment:
Once you rebase, we have new func 'NewErrUnexpectedTypeConversion' to wrap all the internal errors.